src/index.d.ts

```diff
@@ -14,6 +14,30 @@ declare module 'bootstrap-vue' {
 
   export default VuePlugin
 
+  type TableVariant = 'active' | 'success' | 'info' | 'warning' | 'danger' | 'primary' | 'secondary' | 'light' | 'dark'
+  type SortDirection = 'asc' | 'desc'
+  type FormatterCallback = ((value: mixed, key: string, item: mixed) => string)
+
+  export interface TableField {
+    label?: string
+    headerTitle?: string
+    headerAbbr?: string
+    class?: string | string[]
+    formatter?: string | FormatterCallback
+    sortable: boolean
+    sortDirection: SortDirection
+    tdClass: string | string[] | FormatterCallback
+    thClass: string | string[]
+    thStyle: mixed
+    variant: TableVariant
```

Comment:
Maybe update to the following:

```suggestion
    variant: TableVariant | string
```